examples/queue.rs

```diff
@@ -0,0 +1,91 @@
+/// FreeRTOS queue example
+/// The core concepts covered in this example include:
+/// - Creating and using queues to exchange data between an ISR(Interrupt Service Routine) and a thread (both directions).
+/// - Efficiently waiting for messages in the thread using queues.
+///
+/// This is demonstrated with a periodic timer ISR and the rust main thread, but it can be used in any thread or ISR context.
+use esp_idf_hal::delay::BLOCK;
+use esp_idf_hal::peripherals::Peripherals;
+use esp_idf_hal::task;
+use esp_idf_hal::task::queue::Queue;
+use esp_idf_hal::timer::{TimerConfig, TimerDriver};
+use esp_idf_sys::{uxQueueMessagesWaiting, EspError};
+
+fn main() -> Result<(), EspError> {
+    // It is necessary to call this function once. Otherwise some patches to the runtime
+    // implemented by esp-idf-sys might not link properly. See https://github.com/esp-rs/esp-idf-template/issues/71
+    esp_idf_hal::sys::link_patches();
+
+    let per = Peripherals::take()?;
+
+    let mut timer = TimerDriver::new(per.timer00, &TimerConfig::new().auto_reload(true))?;
+
+    // create two queue's that can hold 20/100 elements with the element size of an u8
+    let queue_send = Queue::<u8>::new(20);
+    let queue_recv = Queue::<u8>::new(100);
+
+    // SAFETY: as long as queue_send/recv live we can use isr_receive & isr_send.
+    let isr_recv: Queue<u8> = unsafe { Queue::new_borrowed(queue_send.as_raw()) };
```

Comment:
Nit - as per my comment in the main PR thread, this is just not necessary. Just borrowing the queues with e.g. `&queue_send` and `&queue_recv` is good enough.